src/transformers/models/wav2vec2/feature_extraction_wav2vec2.py

```diff
@@ -205,6 +202,9 @@ def __call__(
             padded_input_values = padded_inputs["input_values"]
             input_lengths = [padded_input_values.shape[-1] for _ in range(padded_input_values.shape[0])]
 
+        if isinstance(padded_inputs["input_values"][0], np.ndarray):
```

Comment:
My plan was to convert as early as possible from `np.float64` to `np.float32` here to save memory and speed, but we should look into making this better anyways! See: https://github.com/huggingface/transformers/issues/12862

Reply:
I actually shifted it from `zero_mean_unit_var_norm()` to here just to fix the case when `do_normalize=False` and the arrays don't end up being converted to `float32`

Reply:
not sure this works tbh we have to be careful that normalization still works as expected with truncation and padding